### DISCUSSION TITLE
Wanting to Contribute

### DISCUSSION THREAD

**Discussion by PaleHazy** [getseabird/seabird#12] [General]
This project looks absolutely like a perfect match for the things I am studying right now. I was wondering if maybe you had some tasks you wanted to delegate?

**Comment by jgillich** [comment]
Heya, there's still plenty of columns and properties to be added on many resources, implementing these isn't too hard I hope. One example, I think I'd be nice to display CPU and memory usage as a column for nodes, similar to Lens. We have this for containers currently:

![Screenshot from 2024-01-28 23-46-53](https://github.com/getseabird/seabird/assets/347965/9be243d6-d5d4-45fb-97a8-2127864f213f)

(Lens isn't displaying any metrics because it doesn't support metrics-server, but I hope the intention is clear :smile: )

The code for these widgets is [here](https://github.com/getseabird/seabird/blob/dca33fb5daf6102363103ab6f11249725f956a44/ui/detail_view.go#L277), it probably wouldn't hurt to extract them as an abstract widget into the /widget directory. Columns are added [here](https://github.com/getseabird/seabird/blob/main/ui/list_view.go#L123). To fetch the `NodeMetrics`, you can put a function on ListBehavior

**Reply by PaleHazy** [reply]
very clear. I just need some assistance setting my dev environment up

**Reply by jgillich** [reply]
Which platform are you on? You need Go plus the dependencies listed in the readme. On Linux and Mac, you can install everything with your package manager. Windows is a bit more difficult, I'd probably use msys2 there.

**Reply by PaleHazy** [reply]
I was on windows, went through many steps and many installs with msys2. But failed at the build command. So now I switched to ubuntu22.04.3 desktop fresh install. And ran this: 

`sudo apt install libgtk-4-dev libgtksourceview-5-dev libadwaita-1-dev libgirepository1.0-dev libglib2.0-dev-bin golang-go`

I get this error when trying to make a build:
```
# github.com/diamondburned/gotk4-adwaita/pkg/adw
../../../go/pkg/mod/github.com/diamondburned/gotk4-adwaita/pkg@v0.0.0-20240107232253-5f283b5d35bc/adw/adw-about-window.go:22:10: could not determine kind of name for C.AdwAboutWindowClass
# github.com/diamondburned/gotk4-sourceview/pkg/gtksource/v5
../../../go/pkg/mod/github.com/diamondburned/gotk4-sourceview/pkg@v0.0.0-20240109205950-92962b56dd0f/gtksource/v5/gtksourcecompletioncontext.go:311:10: could not determine kind of name for C.gtk_source_completion_context_get_proposals_for_provider
../../../go/pkg/mod/github.com/diamondburned/gotk4-sourceview/pkg@v0.0.0-20240109205950-92962b56dd0f/gtksource/v5/gtksourcecompletioncontext.go:389:10: could not determine kind of name for C.gtk_source_completion_context_list_providers

```

am I doing something bad ? is it the os ?

**Reply by jgillich** [reply]
Yes, we need libadwaita 1.4, Ubuntu 22.04 ships an older version. Try Ubuntu 23.10

Or, on Windows, make sure you have these deps: https://github.com/getseabird/seabird/blob/b217706c5300aef03dbee65a674cfc2d9eb9bcb2/.github/workflows/release.yaml#L100-L107

Then run go through msys2. I've done this on my Windows box too, it should build without problems. But I'm not sure if it's possible to get IDE integration with msys, Linux is definitely easier :smile:

**Reply by PaleHazy** [reply]
So latest Ubuntu worked like a charm, I'll carry on there. Thank you!
